index.js

```diff
@@ -138,8 +138,8 @@ HtmlWebpackPlugin.prototype.apply = function (compiler) {
       .then(function (html) {
         var pluginArgs = {html: html, assets: assets, plugin: self, outputName: self.childCompilationOutputName};
         return applyPluginsAsyncWaterfall('html-webpack-plugin-before-html-processing', pluginArgs)
-          .then(function () {
-            return pluginArgs.html;
+          .then(function (newPluginArgs) {
+            return newPluginArgs && newPluginArgs.html || pluginArgs.html;
```

Comment:
> Shouldn't this be newPluginArgs ? newPluginArgs.html : pluginArgs.html

It's the same, or am I missing something?

> Can you please add a unit test for this?

Sure! And I'm going to test if it can be changed by more than one plugin